src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -3,17 +3,49 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int ARRAY_LENGTH = 10;
+    private K key;
+    private V value;
+    private int size;
+    private StorageImpl<K, V>[] arrayStorage;
```

Comment:
Let's create class `Element<K, V>` with fields `K key`, `V value` and use it instead of `StorageImpl`
```suggestion
    private Element<K, V>[] storage;
```